logging/logging-processor/src/main/java/org/infinispan/logging/processor/InfinispanLoggingProcessor.java

```diff
@@ -0,0 +1,127 @@
+package org.infinispan.logging.processor;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic;
+import javax.tools.StandardLocation;
+
+import org.infinispan.logging.processor.report.XmlReportWriter;
+import org.jboss.logging.processor.apt.ProcessingException;
+import org.kohsuke.MetaInfServices;
+/**
+ * InfinispanLoggingProcessor.
+ * @author Durgesh Anaokar
+ * @Since 13.0
+ */
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+@SupportedAnnotationTypes({InfinispanLoggingProcessor.DESCRIPTION})
+@MetaInfServices(Processor.class)
+public class InfinispanLoggingProcessor extends AbstractProcessor {
+   static final String DESCRIPTION = "org.infinispan.logging.annotations.Description";
+   static final String MESSAGE = "org.jboss.logging.annotations.Message";
+   static final String LOGMESSAGE = "org.jboss.logging.annotations.LogMessage";
+
+   @Override
+   public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+      for (TypeElement annotation : annotations) {
+         try {
+            createReportXml(roundEnv, annotation);
+         } catch (Throwable t) {
+            uncaughtException(t);
+            t.printStackTrace();
+            return false;
+         }
+      }
+      return true;
+   }
+
+   private void createReportXml(RoundEnvironment roundEnv, TypeElement annotation) throws IOException {
+      try {
+         Set<? extends Element> annotatedElements = roundEnv.getElementsAnnotatedWith(annotation);
+         Map<String, List<Element>> mapWithQualifiedName = getMapWithQualifiedName(annotatedElements);
```

Comment:
I don't think we need `getMapWithQualifiedName` here. We can make this code simpler by using the [Elements](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/util/Elements.html) interface that's available via `processingEnv.getElementUtils();`.

It should be possible todo something like:

```java
         for (Element e : roundEnv.getElementsAnnotatedWith(annotation)) {
            Elements elements = processingEnv.getElementUtils();
            Element enclosing = e.getEnclosingElement();
            String packageName = elements.getPackageOf(enclosing).getQualifiedName().toString();
            String simpleName = enclosing.getSimpleName().toString();
            ...
         }
```